### ISSUE TITLE
armsubscriptions newListPager doesn't return all subscriptions 

### ISSUE THREAD

**Issue by rajholla** [Azure/azure-sdk-for-go#22279]
### Bug Report

- import path of package in question, e.g. `.../services/compute/mgmt/2018-06-01/compute`

github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions 

- SDK version e.g. `master`, `latest`, `18.1.0`
  - Specify the exact commit if possible; one way to get this is the REVISION
    column output by `go list -m <module>`, for example `go list -m github.com/Azure/azure-sdk-for-go/sdk/azcore`.

github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions v1.3.0

- output of `go version`
tested in both `go version go1.20.5 darwin/arm64` and `go version go1.21.6 linux/arm64`

- What happened?
Having trouble pulling all subscription with in a tenant using below sample code 

```go
cred, err := azidentity.NewDefaultAzureCredential(nil)
	if err != nil {
		log.Fatalf("failed to obtain a credential: %v", err)
	}

	clientFactory, err := armsubscriptions.NewClientFactory(cred, nil)
	if err != nil {
		log.Fatalf("failed to create client: %v", err)
	}
	pager := clientFactory.NewClient().NewListPager(nil)
	totals := make([]string, 0)
	totalPages := 0
	for pager.More() {
		ctx := context.Background()
		page, err := pager.NextPage(ctx)
		if err != nil {
			log.Fatalf("failed to advance page: %v", err)
		}
		for _, v := range page.Value {
			totals = append(totals, *v.ID)
		}
		totalPages++
	}
	log.Println("total subscriptions:", len(totals), totalPages)
```

- What did you expect or want to happen?
Expected to see all subscription. Above code doesn't move to second page. Interestingly, when version of `armsubscription` is downgraded to v1.1.0 , it returns all subscriptions. 

doesn't work -- > github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions v1.3.0
works  -- > github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions v1.1.0

- How can we reproduce it?
above code with `github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions v1.3.0` should be able to reproduce. Make sure to choose tenant that has subscriptions beyond single page.

- Anything we should know about your environment.
I was able to test this behavior on newly built container to rule out potential local conflicts.

**Comment by lirenhe** [comment]
@Alancere , could you try to repo this issue to confirm whether this is a regression？

**Comment by Alancere** [comment]
Hi @rajholla, could you try again? armsubscriptions/v1.2.0 and armsubscriptions/v1.3.0 NewListPager use the same api-version: 2022-12-01

**Comment by rajholla** [comment]
@Alancere  @lirenhe 
This issue was limited to 19th & 20th Jan 2023. I was able to pinpoint the issue to RestAPIs.
Subscriptions: List API returned incomplete results when the latest API version was used ('2022-12-01'). It returned the correct results when the previous version was used, '2021-01-01'.

I have opened a ticket with Azure support. Awaiting RCA.
Feel free to close the issue since it is not related to SDK.
